Microsoft.Toolkit.Uwp.UI.Animations/CompositionAnimations/AnimationCollection.cs

```diff
@@ -0,0 +1,137 @@
+// ******************************************************************
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Collections.Specialized;
+using System.Linq;
+using Windows.UI.Composition;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Hosting;
+
+namespace Microsoft.Toolkit.Uwp.UI.Animations
+{
+    /// <summary>
+    /// An ObservableCollection of <see cref="AnimationBase"/>
+    /// </summary>
+    public class AnimationCollection : ObservableCollection<AnimationBase>
+    {
+        internal UIElement Element { get; set; }
+
+        private void AnimationChanged(object sender, EventArgs e)
+        {
+            AnimationCollectionChanged?.Invoke(this, null);
+        }
+
+        /// <inheritdoc/>
+        protected override void OnCollectionChanged(NotifyCollectionChangedEventArgs e)
+        {
+            base.OnCollectionChanged(e);
+
+            if (e.Action == NotifyCollectionChangedAction.Move)
+            {
+                return;
+            }
+
+            if (e.NewItems != null)
+            {
+                foreach (AnimationBase newAnim in e.NewItems)
+                {
+                    newAnim.AnimationChanged += AnimationChanged;
+                }
+            }
+
+            if (e.OldItems != null)
+            {
+                foreach (AnimationBase oldAnim in e.OldItems)
+                {
+                    oldAnim.AnimationChanged -= AnimationChanged;
+                }
+            }
```

Comment:
Reset is never handled. In a reset event, old and new items are null, so you'll never unhook/rehook

Reply:
Ah, good catch

Reply:
Moved to extend IList instead and handle this in the individual methods